File: 0066-plus-one/0066-plus-one.java
class Solution {
    public int[] plusOne(int[] digits) {
        int carry = 0;
        for(int i = digits.length-1; i >= 0; i--){
            if(digits[i] <= 8){
                digits[i]++;
                carry = 0;
                break;
            }else {
                digits[i] = 0;
                carry = 1;
            }
        }
        if(carry == 1){
            digits = new int[digits.length+1];
            digits[0] = 1;
        }
        return digits;
    }
}

File: 0066-plus-one/0066-plus-one.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        bool carry = 0;
        for(int i = digits.size()-1; i >= 0; i--){
            if(digits[i] <= 8) {
                digits[i] += 1;
                carry = 0;
                break;
            }else {
                digits[i] = 0;
                carry = 1;
            }
        }
        if(carry){
            digits.insert(digits.begin(), 1);
        }
        return digits;
    }
};